### PR TITLE
Docs: update readme, add docs/readme, modernize a bit

### PR DESCRIPTION
**Pull request by ebidel**
- Updates screenshots
- moves main readme examples to docs/readme.md
- adds a "turn on logging" example to the node snippet
- Beefs up the custom audit recipe and removes dupe content from the readme
- Adds info on DevTools integration
- Adds recipe/doc links to main readme

### CODE REVIEW

File: docs/readme.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+Useful documentation, examples, and [recipes](./recipes/) to get you started.
```
Comment (paulirish) [inline docs/readme.md]: kinda think this should be `overview.md` or `basics.md`.

This kinda forces the user to choose a markdown file when entering /docs but i think that's good. it also means we don't have to maintain this file as a table-of-contents of all our docs

**Comment by paulirish** [comment]
These changes look great. Nice stuff. :)

File: readme.md [hunk 2]
```diff
@@ -128,71 +136,13 @@ NOTE: specifying an output path with multiple formats ignores your specified ext
...
-```
-
-### Recipes
-> Helpful for CI integration
-
-  - [gulp](docs/recipes/gulp)
-
 ## Viewing a report
 
 Lighthouse can produce a report as JSON, HTML, or stdout CLI output.
```
Comment (ebidel) [inline readme.md]: good catch. done

File: readme.md [hunk 3]
```diff
@@ -207,17 +157,10 @@ right corner and signing in to GitHub.
...
-* [webpack-lighthouse-plugin](https://github.com/addyosmani/webpack-lighthouse-plugin) -  run Lighthouse from a Webpack build.
-* [lighthouse-mocha-example](https://github.com/justinribeiro/lighthouse-mocha-example) -  gathers performance metrics via Lighthouse and tests them in Mocha
-* [pwmetrics](https://github.com/paulirish/pwmetrics/) - gather performance metrics
-* [lighthouse-hue](https://github.com/ebidel/lighthouse-hue) - Lighthouse score setting the color of Philips Hue lights
-* [lighthouse-batch](https://www.npmjs.com/package/lighthouse-batch) - Run Lighthouse over a number of sites in sequence and generating a summary report including all of their scores.
-* [lighthouse-cron](https://github.com/thearegee/lighthouse-cron) - Cron multiple batch Lighthouse audits and emit results for sending to remote server.
-
 ## Develop
 
+Also see [Contributing](./CONTRIBUTING.md) for more information.
```
Comment (ebidel) [inline readme.md]: done

File: docs/readme.md (hunk 1 above)
Comment (ebidel) [inline docs/readme.md]: I actually for that very reason :)...so people would see some of the most common examples when they hit /docs. It's not really a table of contents though. `[recipes](./recipes/)` will give them the list of directories/md files.

Comment (paulirish) [inline docs/readme.md]: well... im okay with it being `readme`. but let's add one more link to `architecture.md` at the top or so.

File: readme.md [hunk 4]
```diff
@@ -1,32 +1,40 @@
...
 ## Running Lighthouse
 
+### Chrome DevTools
+
+As of Chrome 60 or later, Lighthouse is integrated directly into the Chrome DevTools.
+
+To use Lighthouse from within the DevTools, open the tools, select the Audits panel,
+and hit "Perform an Audit...".
+
+<img width="350px" alt="Lighthouse integration in CHrome DevTools" src="https://cloud.githubusercontent.com/assets/238208/26366636/ada298f8-3fa0-11e7-9da5-ede2c906d10c.png">
```
Comment (patrickhulce) [inline readme.md]: nit: s/CHrome/Chrome/ :)

File: docs/readme.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+launchChromeAndRunLighthouse('https://example.com', flags).then(...);
+```
+
+## Testing on a site with authentication
+
+Lighthouse adds `chrome-debug` to the CLI when you install it from npm.
+
+- Run `chrome-debug`
+- open and login to your site
```
Comment (brendankenny) [inline docs/readme.md]: "navigate to and log in to your site"?

File: readme.md [hunk 6]
```diff
@@ -207,17 +157,11 @@ right corner and signing in to GitHub.
...
-* [lighthouse-cron](https://github.com/thearegee/lighthouse-cron) - Cron multiple batch Lighthouse audits and emit results for sending to remote server.
-
 ## Develop
 
+Read on for the basics of hacking on Lighthouse. Also see [Contributing](./CONTRIBUTING.md)
+for detailed information.
+
 ### Setup
 
 ```sh
```
Comment (brendankenny) [inline readme.md]: will you ditch the comma before "first" below

File: readme.md [hunk 7]
```diff
@@ -234,8 +178,6 @@ yarn build-all
 # cd lighthouse-cli && yarn dev
```
Comment (brendankenny) [inline readme.md]: does `chrome-launcher` need a `tsc` note too?

File: readme.md [hunk 8]
```diff
@@ -312,50 +211,43 @@ yarn closure
...
-      }
-    }]
-  }]
-}
-```
+## Docs & Recipes
+
+Useful documentation, examples, and recipes to get you started.
+
+**Docs**
```
Comment (brendankenny) [inline readme.md]: should this be moved higher? Not as useful for first time visitors, but useful for repeat visitors. At least, when I need to get to a particular set of instructions I usually go to a project's readme and hope that it's linked to from there. Don't know what other people do :)

File: readme.md [hunk 9]
```diff
@@ -312,50 +211,43 @@ yarn closure
...
+The session from Google I/O 2017 covers architecture, writing custom audits,
+Github/Travis/CI integration, headless Chrome, and more:
+
+[![Lighthouse @ Google I/O](https://img.youtube.com/vi/NoRYn6gOtVo/0.jpg)](https://www.youtube.com/watch?v=NoRYn6gOtVo)
+
+_click to watch the video_
+
+## Related Projects
+
+* [webpack-lighthouse-plugin](https://github.com/addyosmani/webpack-lighthouse-plugin) -  run Lighthouse from a Webpack build.
```
Comment (brendankenny) [inline readme.md]: irregular spacing here and below?

Comment (brendankenny) [inline readme.md]: we should fix up the wording on these, too. "gather" vs "gathers", active vs passive, etc

File: readme.md [hunk 10]
```diff
@@ -312,50 +211,43 @@ yarn closure
...
+* [lighthouse-mocha-example](https://github.com/justinribeiro/lighthouse-mocha-example) -  gathers performance metrics via Lighthouse and tests them in Mocha
+* [pwmetrics](https://github.com/paulirish/pwmetrics/) - gather performance metrics
+* [lighthouse-hue](https://github.com/ebidel/lighthouse-hue) - Lighthouse score setting the color of Philips Hue lights
+* [lighthouse-batch](https://www.npmjs.com/package/lighthouse-batch) - Run Lighthouse over a number of sites in sequence and generating a summary report including all of their scores.
```
Comment (brendankenny) [inline readme.md]: generating -> generate? Or Runs/generates. Depends on how the others end up

File: readme.md [hunk 11]
```diff
@@ -374,13 +266,22 @@ If you'd like to contribute, check the [list of issues](https://github.com/Googl
 Nope. Lighthouse runs locally, auditing a page using a local version of the Chrome browser installed the
 machine. Report results are never processed or beaconed to a remote server.
 
-### Videos
+### How do I author custom audits to extend Lighthouse?
 
-Our session from Google I/O 2017: architecture, writing custom audits, Github/Travis/CI integration, and more:
+> **Tip**: see [Lighthouse Architecture](./docs/architecture.md) more information
```
Comment (brendankenny) [inline readme.md]: "see Lighthouse Architecture for information on terminology and architecture"?

File: readme.md [hunk 12]
```diff
@@ -247,50 +189,7 @@ through the entire app. See [Debugging Node.js with Chrome
...
-      ]
-    }
-  ]
-}
-```
-
-## Tests
+### Tests
 
 Some basic unit tests forked are in `/test` and run via mocha. eslint is also checked for style violations.
```
Comment (brendankenny) [inline readme.md]: this sentence has morphed into something that makes no sense (and the "basic unit tests" doesn't apply anymore...lots of unit tests plus smoke tests now)

File: docs/readme.md [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+The example below shows how to run Lighthouse programmatically as a Node module. It
+assumes you've installed Lighthouse as a dependency (`yarn add --dev lighthouse`).
+
+```javascript
+const lighthouse = require('lighthouse');
+const chromeLauncher = require('lighthouse/chrome-launcher/chrome-launcher');
+
+function launchChromeAndRunLighthouse(url, flags, config = null) {
+  return chromeLauncher.launch().then(chrome => {
+    flags.port = chrome.port;
```
Comment (paulirish) [inline docs/readme.md]: i think it's worth it. but if you wanna discuss, let's open an issue?

File: docs/readme.md [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+
+const flags = {logLevel: 'info', output: 'json'};
+log.setLevel(flags.logLevel);
+
+launchChromeAndRunLighthouse('https://example.com', flags).then(...);
+```
+
+## Testing on a site with authentication
+
+Lighthouse adds `chrome-debug` to the CLI when you install it from npm.
```
Comment (paulirish) [inline docs/readme.md]: > When installed globally via `npm install -g` or `yarn global add` a `chrome-debug` binary is added, which will launch a standalone Chrome instance, with a debugging port open.

File: docs/readme.md [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+        "critical-request-chains": { "expectedValue": 0, "weight": 1}
+      }
+    }]
+  }]
+}
+```
+
+Then, run with: `lighthouse --config-path=config.json http://www.random.url`
+
+The traceviewer-based trace processor from [node-big-rig](https://github.com/GoogleChrome/node-big-rig/tree/master/lib) was forked into Lighthouse. Additionally, the [DevTools' Timeline Model](https://github.com/paulirish/devtools-timeline-model) is available as well. There may be advantages for using one model over another.
```
Comment (paulirish) [inline docs/readme.md]: sg. let's trash this bit.

File: readme.md [hunk 16]
```diff
@@ -1,32 +1,40 @@
...
+# Lighthouse  [![Build Status](https://travis-ci.org/GoogleChrome/lighthouse.svg?branch=master)](https://travis-ci.org/GoogleChrome/lighthouse) [![Coverage Status](https://coveralls.io/repos/github/GoogleChrome/lighthouse/badge.svg?branch=master)](https://coveralls.io/github/GoogleChrome/lighthouse?branch=master) [![NPM lighthouse package](https://img.shields.io/npm/v/lighthouse.svg)](https://npmjs.org/package/lighthouse)
 
-> Lighthouse analyzes web apps and web pages, collecting modern performance metrics and insights on developer best practices.
-
-**Lighthouse requires Chrome 56 or later.**
+> [Lighthouse](https://developers.google.com/web/tools/lighthouse/) analyzes web apps and web pages, collecting modern performance metrics and insights on developer best practices.
 
 ## Installation
 
+_Lighthouse requires Chrome 56 or later._
```
Comment (paulirish) [inline readme.md]: Ya. let's do this. can link to https://googlechrome.github.io/current-versions/

File: docs/readme.md (hunk 13 above)
Comment (brendankenny) [inline docs/readme.md]: I don't know how much it hurts things or how much the gains matter, so not sure if it's worth opening an issue :)

It's just when I see examples like this it seems like one more requirement to remember without any advantages for basic usage

File: readme.md (hunk 4 above)
Comment (ebidel) [inline readme.md]: done

Comment (ebidel) [inline readme.md]: done

File: readme.md (hunk 12 above)
Comment (ebidel) [inline readme.md]: done

File: docs/readme.md [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+      chrome.kill().then(() => results)
+    );
+  });
+}
+
+const flags = {output: 'json'};
+
+// Usage:
+launchChromeAndRunLighthouse('https://example.com', flags)
+  .then(results => console.log(results));
```
Comment (ebidel) [inline docs/readme.md]: done

File: docs/readme.md [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+
+If you want to see log output as Lighthouse runs, include the `log` module
+and set an appropriate logging level in your code. You'll also need to pass
+the `logLevel` flag when calling `lighthouse`.
+
+```javascript
+const log = require('lighthouse/lighthouse-core/lib/log');
+
+const flags = {logLevel: 'info', output: 'json'};
+log.setLevel(flags.logLevel);
```
Comment (ebidel) [inline docs/readme.md]: you need it to see the chrome launcher's output.

File: docs/readme.md (hunk 14 above)
Comment (ebidel) [inline docs/readme.md]: done

File: docs/readme.md (hunk 5 above)
Comment (ebidel) [inline docs/readme.md]: done

File: docs/readme.md [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+00a2fd8b1e631fcb       device usb:335682009X product:bullhead model:Nexus_5X device:bullhead
+
+$ adb forward tcp:9222 localabstract:chrome_devtools_remote
+
+$ lighthouse --disable-device-emulation --disable-cpu-throttling https://mysite.com
+```
+
+## Lighthouse as trace processor
+
+Lighthouse can be used to analyze trace and performance data collected from other tools (like WebPageTest and ChromeDriver). The `traces` and `devtoolsLogs` artifact items can be provided using a string for the absolute path on disk. The `devtoolsLogs` array is captured from the Network domain (a la ChromeDriver's [`enableNetwork` option](https://sites.google.com/a/chromium.org/chromedriver/capabilities#TOC-perfLoggingPrefs-object)) and reformatted slightly. As an example, here's a trace-only run that's reporting on user timings and critical request chains:
```
Comment (ebidel) [inline docs/readme.md]: So what are are the updates?

File: docs/readme.md (hunk 15 above)
Comment (ebidel) [inline docs/readme.md]: removed

File: docs/readme.md (hunk 18 above)
Comment (brendankenny) [inline docs/readme.md]: > you need it to see the chrome launcher's output

that seems like a chrome-launcher bug. I think we discussed removing `log` from there, but it should take a `logLevel` until/if it gets removed

File: docs/recipes/custom-audit/readme.md [hunk 20]
```diff
@@ -1,11 +1,34 @@
-# Basic custom audit recipe for Lighthouse
+# Basic Custom Audit Recipe
 
-A hypothetical site measures the time from navigation start to when the page has initialized and the main search box is ready to be used. It saves that value in a global variable, `window.myLoadMetrics.searchableTime`.
+> **Tip**: see [Lighthouse Architecture](../../../docs/architecture.md) more information
```
Comment (ebidel) [inline docs/recipes/custom-audit/readme.md]: done

File: docs/recipes/custom-audit/readme.md [hunk 21]
```diff
@@ -1,11 +1,34 @@
...
+on terminology and architecture.
 
-This Lighthouse [gatherer](searchable-gatherer.js)/[audit](searchable-audit.js) pair will take that value from the context of the page and test whether or not it stays below a test threshold.
+## What this example does
 
-The config file tells Lighthouse where to find the gatherer and audit files, when to run them, and how to incorporate their output into the Lighthouse report.
+This example shows how to write a custom Lighthouse audit for a hypothetical site
+which measures the time from navigation start to when the page has initialized.
+The page is considered fully initialized when the main search box ("hero element")
+is ready to be used. The page saves that time in a global variable called `window.myLoadMetrics.searchableTime`.
```
Comment (ebidel) [inline docs/recipes/custom-audit/readme.md]: done

File: docs/recipes/custom-audit/readme.md [hunk 22]
```diff
@@ -1,11 +1,34 @@
...
+from the context of the page.
+
+[searchable-audit.js](searchable-audit.js) - an [Audit](https://github.com/GoogleChrome/lighthouse/blob/master/docs/architecture.md#components--terminology) that tests whether or not `window.myLoadMetrics.searchableTime`
+stays below a 4000ms. In other words, Lighthouse will consider the audit "passing"
+in the report if the search box initializes within 4s.
+
+[custom-config.js](custom-config.js) - this file tells Lighthouse where to
+find the gatherer and audit files, when to run them, and how to incorporate their
+output into the Lighthouse report. In this example, we have extended [Lighthouse's
+default configuration](https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/config/default.js). If you're extending Lighthouse's default, passes with the same name are merged together, all other arrays will be concatenated, and primitive values will override the defaults.
```
Comment (ebidel) [inline docs/recipes/custom-audit/readme.md]: It was moved from a section in the main readme. Removed.

File: readme.md [hunk 23]
```diff
@@ -1,32 +1,40 @@
-# Lighthouse  [![Build Status](https://travis-ci.org/GoogleChrome/lighthouse.svg?branch=master)](https://travis-ci.org/GoogleChrome/lighthouse) [![Coverage Status](https://coveralls.io/repos/github/GoogleChrome/lighthouse/badge.svg?branch=master)](https://coveralls.io/github/GoogleChrome/lighthouse?branch=master)
+# Lighthouse  [![Build Status](https://travis-ci.org/GoogleChrome/lighthouse.svg?branch=master)](https://travis-ci.org/GoogleChrome/lighthouse) [![Coverage Status](https://coveralls.io/repos/github/GoogleChrome/lighthouse/badge.svg?branch=master)](https://coveralls.io/github/GoogleChrome/lighthouse?branch=master) [![NPM lighthouse package](https://img.shields.io/npm/v/lighthouse.svg)](https://npmjs.org/package/lighthouse)
 
-> Lighthouse analyzes web apps and web pages, collecting modern performance metrics and insights on developer best practices.
-
-**Lighthouse requires Chrome 56 or later.**
+> [Lighthouse](https://developers.google.com/web/tools/lighthouse/) analyzes web apps and web pages, collecting modern performance metrics and insights on developer best practices.
```
Comment (ebidel) [inline readme.md]: meh. We have the link at the top of the repo and sprinkled elsewhere.

File: readme.md [hunk 24]
```diff
@@ -1,32 +1,40 @@
...
 
-### Node CLI [![NPM lighthouse package](https://img.shields.io/npm/v/lighthouse.svg)](https://npmjs.org/package/lighthouse)
+### Node CLI
 
-**Requires Node v6+.**
+_Lighthouse requires Node 6 or later._
 
 ```sh
 npm install -g lighthouse
-# or use yarn:
```
Comment (ebidel) [inline readme.md]: reverted

File: readme.md [hunk 25]
```diff
@@ -1,32 +1,40 @@
...
+
+To use Lighthouse from within the DevTools, open the tools, select the Audits panel,
+and hit "Perform an Audit...".
+
+<img width="350px" alt="Lighthouse integration in CHrome DevTools" src="https://cloud.githubusercontent.com/assets/238208/26366636/ada298f8-3fa0-11e7-9da5-ede2c906d10c.png">
+
 ### Chrome extension
 
-Check out the quick-start guide: <http://bit.ly/lighthouse-quickstart>
+Check out the [quick-start guide](https://developers.google.com/web/tools/lighthouse/#extension).
```
Comment (ebidel) [inline readme.md]: What's the extension quick start guide?

File: readme.md (hunk 7 above)
Comment (ebidel) [inline readme.md]: done

File: readme.md (hunk 25 above)
Comment (brendankenny) [inline readme.md]: Isn't that what that link is? I just meant the link text

File: docs/readme.md (hunk 19 above)
Comment (paulirish) [inline docs/readme.md]: > The `devtoolsLogs` array is captured from the Network domain

to 

> The `devtoolsLogs` array is captured from the Network and Page domains (a la ChromeDriver's [`enableNetwork` and `enablePage ` options]

File: readme.md (hunk 16 above)
Comment (paulirish) [inline readme.md]: >  _Lighthouse requires [Chrome stable](https://googlechrome.github.io/current-versions/) or later._

File: readme.md [hunk 26]
```diff
@@ -1,32 +1,40 @@
...
 npm install -g lighthouse
-# or use yarn:
 # yarn global add lighthouse
 ```
 
 ## Running Lighthouse
 
+### Chrome DevTools
+
+As of Chrome 60 or later, Lighthouse is integrated directly into the Chrome DevTools.
```
Comment (paulirish) [inline readme.md]: i like this too. i'm equally happy doing this in a followup PR since we're already hitting a lot here.

Comment (ebidel) [inline readme.md]: done

File: docs/recipes/custom-audit/readme.md (hunk 22 above)
Comment (brendankenny) [inline docs/recipes/custom-audit/readme.md]: > It was moved from a section in the main readme

we should probably still include it somewhere. As paltry as it is, it's our only documentation of how extending a config works

File: readme.md (hunk 23 above)
Comment (brendankenny) [inline readme.md]: > meh. We have the link at the top of the repo and sprinkled elsewhere.

should we just drop it from here then? With the link, the introductory sentence kind of undercuts the rest of the readme by saying "here's a place to go if you want to learn more about Lighthouse" rather than just reading the already-open readme :)

File: docs/recipes/custom-audit/readme.md [hunk 27]
```diff
@@ -1,11 +1,33 @@
-# Basic custom audit recipe for Lighthouse
+# Basic Custom Audit Recipe
 
-A hypothetical site measures the time from navigation start to when the page has initialized and the main search box is ready to be used. It saves that value in a global variable, `window.myLoadMetrics.searchableTime`.
+> **Tip**: see [Lighthouse Architecture](../../../docs/architecture.md) information
```
Comment (brendankenny) [inline docs/recipes/custom-audit/readme.md]: for information

File: readme.md [hunk 28]
```diff
@@ -374,13 +259,22 @@ If you'd like to contribute, check the [list of issues](https://github.com/Googl
 Nope. Lighthouse runs locally, auditing a page using a local version of the Chrome browser installed the
 machine. Report results are never processed or beaconed to a remote server.
 
-### Videos
+### How do I author custom audits to extend Lighthouse?
 
-Our session from Google I/O 2017: architecture, writing custom audits, Github/Travis/CI integration, and more:
+> **Tip**: see [Lighthouse Architecture](./docs/architecture.md) more information
```
Comment (brendankenny) [inline readme.md]: also needs for

File: docs/recipes/custom-audit/readme.md [hunk 29]
```diff
@@ -1,11 +1,33 @@
...
+- [searchable-audit.js](searchable-audit.js) - an [Audit](https://github.com/GoogleChrome/lighthouse/blob/master/docs/architecture.md#components--terminology) that tests whether or not `window.myLoadMetrics.searchableTime`
+stays below a 4000ms threshold. In other words, Lighthouse will consider the audit "passing"
+in the report if the search box initializes within 4s.
+
+- [custom-config.js](custom-config.js) - this file tells Lighthouse where to
+find the gatherer and audit files, when to run them, and how to incorporate their
+output into the Lighthouse report. This example extends [Lighthouse's
+default configuration](https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/config/default.js).
+
+**Note**: when extending default.js, passes with the same name are merged together,
```
Comment (brendankenny) [inline docs/recipes/custom-audit/readme.md]: filename might be obscure. Maybe "when extending the default config"?

File: readme.md [hunk 30]
```diff
@@ -302,60 +218,29 @@ yarn test
...
-```
-
-Then, run with: `lighthouse --config-path=config.json http://www.random.url`
+## Related Projects
 
-The traceviewer-based trace processor from [node-big-rig](https://github.com/GoogleChrome/node-big-rig/tree/master/lib) was forked into Lighthouse. Additionally, the [DevTools' Timeline Model](https://github.com/paulirish/devtools-timeline-model) is available as well. There may be advantages for using one model over another.
+* [webpack-lighthouse-plugin](https://github.com/addyosmani/webpack-lighthouse-plugin) - run Lighthouse from a Webpack build.
+* [lighthouse-mocha-example](https://github.com/justinribeiro/lighthouse-mocha-example) - gather performance metrics via Lighthouse and tests them in Mocha
+* [pwmetrics](https://github.com/paulirish/pwmetrics/) - gather performance metrics
+* [lighthouse-hue](https://github.com/ebidel/lighthouse-hue) - uses a Lighthouse score to set the color of Philips Hue lights
```
Comment (brendankenny) [inline readme.md]: s/uses/use to match the other descriptions

File: readme.md [hunk 31]
```diff
@@ -302,60 +218,29 @@ yarn test
...
+* [lighthouse-batch](https://www.npmjs.com/package/lighthouse-batch) - run Lighthouse over a number of sites and generate a summary of their metrics/scores.
+* [lighthouse-cron](https://github.com/thearegee/lighthouse-cron) - Cron multiple batch Lighthouse audits and emit results for sending to remote server.
```
Comment (brendankenny) [inline readme.md]: the start of this is hard to parse, but that's what was submitted so we can leave as is for now :)
